packages/web/stores/assets/assets-store.ts

```diff
@@ -157,49 +153,53 @@ export class ObservableAssets {
     sourceChainNameOverride?: string;
   })[] {
     return this.assets
-      .filter((asset) => asset.origin_chain_id !== this.chain.chainId) // Filter osmosis native assets
+      .filter((asset) => asset.transferMethods.length > 0) // Filter osmosis native assets
       .filter((asset) => {
         if (typeof window === "undefined") return true;
 
         if (sessionStorage.getItem(UnlistedAssetsKey) === "true") {
           return true;
         }
-        return !asset.keywords?.includes("osmosis-unlisted");
+        return !asset.preview;
       }) // Remove unlisted assets if preview assets is disabled
       .map((ibcAsset) => {
-        const chainInfo = this.chainStore.getChain(ibcAsset.origin_chain_id);
+        const chainInfo = this.chainStore.getChain(
+          ibcAsset.counterparty[0]?.chainId?.toString() ?? ""
+        );
 
-        const minimalDenom = getSourceDenomFromAssetList(ibcAsset);
+        const sourceDenom_ = ibcAsset.sourceDenom;
```

Comment:
How about we rename `sourceDenom_` to `ibcAssetSourceDenom`? I believe it's more descriptive